core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/MetricsServiceImpl.java

```diff
@@ -490,6 +490,22 @@ void setDataAccess(DataAccess dataAccess) {
     }
 
     @Override
+    public <T> Observable<T> findGaugeData(String tenantId, MetricId id, Long start, Long end,
+            Func1<Observable<DataPoint<Double>>, Observable<T>>... funcs) {
+
+        Observable<DataPoint<Double>> dataCache = findGaugeData(tenantId, id, start, end).cache();
+        Observable<T> result = null;
+
+        for (Func1<Observable<DataPoint<Double>>, Observable<T>> func : funcs) {
+            result = (null == result) ?
+                    func.call(dataCache) :
+                    result.concatWith(func.call(dataCache));
+        }
+
+        return result;
+    }
```

Comment:
Excellent point. Only ```flatMap``` call is parallel because of the ```subscribeOn``` call. This would have certainly have to documented in the version of the method that takes the Scheduler argument. The additional method is not necessary for the PR.